### PR TITLE
Add multistage build for rust verification libs

### PR DESCRIPTION
**Pull request by ManuelBilbao**
As the verification libs don't change frequently and it takes a lot of time to compile it, I put it in another stage so it's cached in further docker builds.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,28 +1,31 @@
...
 
+USER root
 WORKDIR /root/alignedlayer
 
-RUN make clean-ffi
-RUN make build-linux
+# Get Ubuntu packages
+RUN apt-get install -y build-essential
+
+COPY . .
```
Comment (JulianGCalderon) [inline Dockerfile:20]: Only cosmos files should be copied, right? (ignoring readmes, scripts, and other files):
- api
- app
- cmd
- proto
- testutil
- tools
- x
- Makefile
- config.yml
- go.mod
- go.sum

**Review by JulianGCalderon — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (ManuelBilbao) [reply Dockerfile:20]: I think add every cosmos file at hand would be dirty. I'm adding unnecessary files to .dockerignore to avoid copying them

Comment (ManuelBilbao) [reply Dockerfile:20]: Done

**Review by JulianGCalderon — APPROVED** [review]
LGTM

**Pull request merged**